WSL/disk-space.md

```diff
@@ -0,0 +1,210 @@
+```
+
+The output will include:
+
+- **Filesystem**: Identifier for the VHD file system
+- **Size**: Total size of the disk (the maximum amount of space allocated to the VHD)
+- **Used**: Amount of space currently being used in the VHD
+- **Avail**: Amount of space left in the VHD (Allocated size minus amount used)
+- **Use%**: Percentage of disk space remaining (Used / Allocated size)
+- **Mounted on**: Directory path where the disk is mounted
+
+If you see that you are near to reaching the available amount of disk space allocated to your VHD, or have already received an error due to no disk space remaining, see the next section for steps on how to expand the maximum amount of disk space allocated to the VHD associated with your Linux distribution.
+
+> [!NOTE]
+> The size of the VHD represented by the output of the df command represents the size allocated for the Linux file system and will be different than the total size on your device used by the Linux distribution. You can view otal disk space used by the distribution by going to **Settings > Apps > Installed apps**, find the Linux distribution name you want to view (ie. "Ubuntu"), select `...` and choose **Advanced options** to view the total disk space used, amount of data stored, and version number of your installed distribution. You can also **Uninstall** the distribution.
+
+## How to expand the size of your WSL 2 Virtual Hard Disk
+
+To expand the VHD size for a Linux distribution beyond the 1TB maximum amount of allocated disk space, follow the steps below. *(For earlier WSL releases that have not yet been updated, this max default may be set to 512GB or 256GB).*
+
+1. Terminate all WSL instances using the command: `wsl --shutdown`
+
+2. Copy the directory path to the *ext4.vhdx* file associated with the Linux distribution installed on your machine. For help, see [How to locate the vhdx file and disk path for your Linux distribution](#how-to-locate-the-vhdx-file-and-disk-path-for-your-linux-distribution).
+
+3. Open Windows Command Prompt with admin privileges and then open the [diskpart](/windows-server/administration/windows-commands/diskpart) command interpreter by entering:
+
+      ```cmd
+      diskpart
+      ```
+
+4. You will now have a `DISKPART>` prompt. Enter the following command, replacing `<pathToVHD>` with the directory path to the `ext4.vhdx` file associated with the Linux distribution (copied in step #2).
+
+      ```cmd
+      Select vdisk file="<pathToVHD>"
+      ```
+
+5. Display the details associated with this virtual disk, including the **Virtual size**, representing the current maximum size the VHD is allocated:
+
+      ```cmd
+      detail vdisk
+      ```
+
+6. You will need to convert the **Virtual size** to megabytes. For example, if **Virtual size: 1024GB**, convert this to **1024000**. The new value you enter must be greater than this original value. For example, to double the virtual size listed above, you could enter the value: **2048000**.
```

Comment:
Are there more parameters we should add here? I tried changing the size of my VHD from 1024000 to 1024010 and received an error. Does the new size have to meet certain criteria?

Reply:
What error did you get ?

Reply:
"DiskPart has encountered an error: The parameter is incorrect."

Reply:
Did you enter 1024010 as megabytes ? That would be a really big size. Does 1025 works ? I think the above section is bit misleading because `1024GB` should be `1024MB`

Reply:
Having a second look at this, should we simply instruct users to use [Resize-Vhd](https://learn.microsoft.com/en-us/powershell/module/hyper-v/resize-vhd?view=windowsserver2022-ps) instead ? It's much simpler to use than diskpart in this case and we've seen users corrupt their disks with diskparts (see https://github.com/microsoft/WSL/issues/9443)

Reply:
We can use whatever you think is best. Could you take a crack at rewriting this part? I've not used Resize-VHD yet.

Reply:
Also, can that Resize-Vhd command be used to reduce/shrink the size of the ext4.vhdx file as well? I've expanded my kali-linux vhd to a large max, but cannot figure out how to return it to the original size. Maybe we should include a bit about this as well?

Reply:
Just pushed a commit on this branch that does that, let me know what you think. I also made some simplification to the resize2fs part, since it can actually detect the correct size on its own

Reply:
Unfortunately, reducing the size of the VHD is much harder operation, since the filesystem needs be shrinked first. If we decide that we need to do that, let's do it in a followup change to unblock this one

Reply:
Added a note to be cautious not to accidentally expand the vhd beyond what you need. Maybe we just see if there are issues that come in asking about reducing vhd size before we add more content to this doc that could potentially be confusing.